src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Symbols/SymbolManagerBase.cs

```diff
@@ -337,16 +331,18 @@ public bool Equals(TypeArrayKey other)
                 if (other._types == _types) return true;
                 if (other._types.Length != _types.Length) return false;
                 if (other._hashCode != _hashCode) return false;
-                for (int i = 0, n = _types.Length; i < n; i++)
+                for (int i = 0; i < _types.Length; i++)
                 {
                     if (!_types[i].Equals(other._types[i]))
                         return false;
                 }
                 return true;
             }
 
+            [ExcludeFromCodeCoverage] // Typed overload should always be the method called.
```

Comment:
I did examine and satisfy myself that there's nothing that could hit it before adding the attribute. The methods could be taken out entirely, but as well as going against good-practice to not override `Equals` when there's a strongly typed version, it'd be easier to catch something that (for good or for ill) hit it due to a later change by having it present and doing `Assert.Fail()`.